src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -52,7 +58,20 @@ internal abstract partial class PatternMatcher : IDisposable
             bool allowFuzzyMatching = false)
         {
             culture = culture ?? CultureInfo.CurrentCulture;
+            if (culture.Name == "en-US")
```

Comment:
> Instead of using ordinal for english and culture-sensitive for other languages, I would prefer an approach that did a fast comparison attempt and fell back to the culture-sensitive comparison if any non-ASCII characters are encountered. This is the approach we took in the System.Refelection.Metadata string comparer: https://source.dot.net/#System.Reflection.Metadata/System/Reflection/Internal/Utilities/MemoryBlock.cs,378

That looks like a heck of a lot of complex code.  Is that avialable in a library that i can try out and measure the perf for? Note that i tried even writing unsafe code, and it looked like calling into the low level helper was *still* faster (possibly due to the native code being better optimized by the runtime).